scripts/project_file_header_fix.sh

```diff
@@ -0,0 +1,224 @@
+            if [ ! -z "$code_file_found_location" ]; then
+                if [[ ! $code_file_found_location -ef $destiny$code_file_basename ]]; then
+                    # Move associated implementation file.
+                    mv -f $code_file_found_location $destiny
+                    echo "File "$code_file" moved to "$destiny
+
+                    # Fix path in the project file.
+                    # Output to a second file, then overwrite the first with the second.
+                    # This is done because sed does not like writing to the file it is currently reading.
+                    sed '/'$code_file_basename'/{s/'$header_opp'/'$header_type'/;}' $project_file > $project_file"2"
+                    mv -f $project_file"2" $project_file
+                else
+                    echo $code_file_found_location" does not need to be moved"
+                fi
+            fi
+        done
+    fi
+
+else
+    echo "All files are in the correct folder..."
+fi
+
+# Find all header files in public/
+public_file_list=$(find "$target_path"public -name '*.h')
+
+if [ ! -z "$public_file_list" ]; then
+    for header_file in $public_file_list
+    do
+        file_basename=$(basename "$header_file")
+        # Use sed to check to see if the file is in the project header
+        found=$(sed -n '/'$file_basename'/{p;}' $project_header)
+        if [ -z "$found" ]; then
+            project_header_not_found_list+=$header_file" "
+        fi
+    done
+fi
+
+# List the files found.
+if [ ! -z "$project_header_not_found_list" ]; then
+    echo
+    echo "These files are public and were not found in "$project_header
+    for header_file in $project_header_not_found_list
+    do
+        echo $header_file
+    done
+    echo "Please add them to the project header, then press enter to continue..."
+    read user_input
+else
+    echo "All public header files were found in "$project_header
+fi
+
+# Find all header files in private/
+private_file_list=$(find "$target_path"private -name '*.h')
+
+if [ ! -z "$private_file_list" ]; then
+    for private_header_file in $private_file_list
+    do
+        file_basename=$(basename "$private_header_file")
+
+        # Use sed to check to see if the file is NOT in the project header
+        found=$(sed -n '/'$file_basename'/{p;}' $project_header)
+        if [ ! -z "$found" ]; then
+            private_file_found_list+=$headprivate_header_fileer_file" "
+        fi
+    done
+fi
+
+# List the files found.
+if [ ! -z "$private_file_found_list" ]; then
+    echo
+    echo "These files are private and were found in "$project_header
+    for private_header_file in $private_file_found_list
+    do
+        echo $private_header_file
+    done
+    echo "Please remove them from the project header, then press enter to continue..."
+    read user_input
+else
+    echo "No private header files were found in "$project_header
+fi
+
+echo "Done checking public / private headers for correctness..."
+read user_input
```

Comment:
```suggestion
echo "Done checking public / private headers for correctness"
```

This one wasn't supposed to have a `read user_input`